lib/ua_update_messaging.py

```diff
@@ -0,0 +1,317 @@
+            )
+            no_pkgs_msg = (
+                MESSAGE_CONTRACT_EXPIRED_APT_NO_PKGS_TMPL.format(
+                    title=ent.title, url=defaults.BASE_ESM_URL
+                )
+                + no_warranty
+            )
+            motd_no_pkgs_msg = no_pkgs_msg
+            motd_pkgs_msg = (
+                MESSAGE_CONTRACT_EXPIRED_MOTD_PKGS_TMPL.format(
+                    title=ent.title,
+                    pkg_num=tmpl_pkg_count_var,
+                    url=defaults.BASE_ESM_URL,
+                )
+                + no_warranty
+            )
+    else:  # Apps not enabled on ESM Ubuntu release
+        pkgs_msg = MESSAGE_DISABLED_APT_PKGS_TMPL.format(
+            title=ent.title,
+            pkg_num=tmpl_pkg_count_var,
+            pkg_names=tmpl_pkg_names_var,
+            url=defaults.BASE_ESM_URL,
+        )
+        no_pkgs_msg = MESSAGE_DISABLED_MOTD_NO_PKGS_TMPL.format(
+            title=ent.title, url=defaults.BASE_ESM_URL
+        )
+        motd_pkgs_msg = MESSAGE_DISABLED_MOTD_PKGS_TMPL.format(
+            title=ent.title,
+            pkg_num=tmpl_pkg_count_var,
+            url=defaults.BASE_ESM_URL,
+        )
+        motd_no_pkgs_msg = MESSAGE_DISABLED_MOTD_NO_PKGS_TMPL.format(
+            title=ent.title, url=defaults.BASE_ESM_URL
+        )
+
+    msg_dir = os.path.join(cfg.data_dir, "messages")
+    _write_template_or_remove(no_pkgs_msg, os.path.join(msg_dir, no_pkgs_file))
+    _write_template_or_remove(pkgs_msg, os.path.join(msg_dir, pkgs_file))
+    _write_template_or_remove(
+        motd_no_pkgs_msg, os.path.join(msg_dir, motd_no_pkgs_file)
+    )
+    _write_template_or_remove(
+        motd_pkgs_msg, os.path.join(msg_dir, motd_pkgs_file)
+    )
+
+
+def write_apt_pre_invoke_templates(cfg: config.UAConfig) -> None:
+    """Write messaging templates about available esm packages.
+
+    :param cfg: UAConfig instance for this environment.
+    """
+    apps_no_pkg_file = ExternalMessage.APT_PRE_INVOKE_APPS_NO_PKGS.value
+    apps_pkg_file = ExternalMessage.APT_PRE_INVOKE_APPS_PKGS.value
+    infra_no_pkg_file = ExternalMessage.APT_PRE_INVOKE_INFRA_NO_PKGS.value
+    infra_pkg_file = ExternalMessage.APT_PRE_INVOKE_INFRA_PKGS.value
+    motd_apps_no_pkg_file = ExternalMessage.MOTD_APPS_NO_PKGS.value
+    motd_apps_pkg_file = ExternalMessage.MOTD_APPS_PKGS.value
+    motd_infra_no_pkg_file = ExternalMessage.MOTD_INFRA_NO_PKGS.value
+    motd_infra_pkg_file = ExternalMessage.MOTD_INFRA_PKGS.value
+
+    apps_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-apps"]
+    apps_inst = apps_cls(cfg)
+    config_allow_beta = util.is_config_value_true(
+        config=cfg.cfg, path_to_value="features.allow_beta"
+    )
+    apps_not_beta = bool(config_allow_beta or not apps_cls.is_beta)
+    infra_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-infra"]
+    infra_inst = infra_cls(cfg)
+
+    expiry_status, remaining_days = get_contract_expiry_status(cfg)
+
+    if apps_not_beta:
```

Comment:
Do we need a `series != "trusty"` check here as well or no?

Reply:
bump to make sure you saw this. I don't know the answer to this

Reply:
Bummer I trashed this change in a prior rebase . Fixing now.